### PR TITLE
[indexer grpc][processor] Add processors to Indexer GRPC system

### PR DESCRIPTION
**Pull request by larry-aptos**
### Description

* Add processors to indexer grpc with starting version.
* Supports restarts

### Test Plan

Tested locally, looks good.

### CODE REVIEW

File: ecosystem/indexer-grpc/indexer-grpc-parser/README.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
+# Aptos Indexer
+
+Will fix readme but just remember command for now
+```
+APTOS_POSTGRES_CONNECTION_STRING_VAR=postgresql://localhost:5432/indexer_v2 APTOS_DATASTREAM_SERVICE_ADDRESS_VAR=http://localhost:50051 STARTING_VERSION=0 cargo run
+```
+
+
```
Comment (bowenyang007) [inline ecosystem/indexer-grpc/indexer-grpc-parser/README.md:8]: Can you delete the rest lol.

File: ecosystem/indexer-grpc/indexer-grpc-parser/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    if std::env::var("CIRCLECI").is_ok() {
+        std::env::var("INDEXER_DATABASE_URL").expect("must set 'INDEXER_DATABASE_URL' in CI!");
+    }
+    if std::env::var("INDEXER_DATABASE_URL").is_ok() {
+        false
+    } else {
+        aptos_logger::warn!("`INDEXER_DATABASE_URL` is not set: skipping indexer tests");
+        true
+    }
+}
```
Comment (bowenyang007) [inline ecosystem/indexer-grpc/indexer-grpc-parser/src/lib.rs:32]: Do we still need this?

File: ecosystem/indexer-grpc/indexer-grpc-utils/src/config.rs [hunk 3]
```diff
@@ -20,30 +20,59 @@ pub struct IndexerGrpcConfig {
     pub health_check_port: u16,
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize, Default)]
+#[serde(deny_unknown_fields)]
+pub struct IndexerGrpcProcessorConfig {
```
Comment (bowenyang007) [inline ecosystem/indexer-grpc/indexer-grpc-utils/src/config.rs:25]: Add some comments on what this is.

Comment (bowenyang007) [reply ecosystem/indexer-grpc/indexer-grpc-utils/src/config.rs:25]: And also some comments on IndexerGrpcConfig.

File: ecosystem/indexer-grpc/indexer-grpc-parser/src/worker.rs [hunk 4]
```diff
@@ -0,0 +1,399 @@
...
+                panic!();
+            },
+        }
+
+        loop {
+            let mut transactions_batches = vec![];
+            // Gets a batch of transactions from the stream. Batch size is set in the grpc server.
+            // The number of batches depends on our config
+            for _ in 0..concurrent_tasks {
+                let next_stream = match resp_stream.next().await {
```
Comment (bowenyang007) [inline ecosystem/indexer-grpc/indexer-grpc-parser/src/worker.rs:193]: Todo: Try next

**Review by bowenyang007 — APPROVED** [review]

File: ecosystem/indexer-grpc/indexer-grpc-parser/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+pub mod models;
+pub mod processors;
+pub mod schema;
+mod utils;
+pub mod worker;
+
+/// By default, skips test unless `INDEXER_DATABASE_URL` is set.
+/// In CI, will explode if `INDEXER_DATABASE_URL` is NOT set.
+pub fn should_skip_pg_tests() -> bool {
+    if std::env::var("CIRCLECI").is_ok() {
```
Comment (gregnazario) [inline ecosystem/indexer-grpc/indexer-grpc-parser/src/lib.rs:23]: CircleCI?

**Review by gregnazario — APPROVED** [review]
Please look into if we need all the migration sql files and some of the other dot files actually committed

File: ecosystem/indexer-grpc/indexer-grpc-parser/README.md (hunk 1 above)
Comment (larry-aptos) [reply ecosystem/indexer-grpc/indexer-grpc-parser/README.md:8]: removed.

File: ecosystem/indexer-grpc/indexer-grpc-parser/src/lib.rs (hunk 2 above)
Comment (larry-aptos) [reply ecosystem/indexer-grpc/indexer-grpc-parser/src/lib.rs:32]: removed; no need for now.

File: ecosystem/indexer-grpc/indexer-grpc-utils/src/config.rs (hunk 3 above)
Comment (larry-aptos) [reply ecosystem/indexer-grpc/indexer-grpc-utils/src/config.rs:25]: done.

File: ecosystem/indexer-grpc/indexer-grpc-parser/src/lib.rs (hunk 5 above)
Comment (larry-aptos) [reply ecosystem/indexer-grpc/indexer-grpc-parser/src/lib.rs:23]: no longer used. removed.

File: ecosystem/indexer-grpc/indexer-grpc-parser/src/worker.rs (hunk 4 above)
Comment (larry-aptos) [reply ecosystem/indexer-grpc/indexer-grpc-parser/src/worker.rs:199]: added todo. thanks!

**Pull request merged**